Makefile

```diff
@@ -49,14 +50,26 @@ docker-build: build-docker-bins
 docker-build: $(addsuffix -image,$(IMAGES))
 
 %-image:
-	docker build $(DOCKER_BUILD_FLAGS) -t $(DOCKER_REGISTRY)/$*:$(IMAGE_TAG) $*
+	docker build $(DOCKER_BUILD_FLAGS) -t $(DOCKER_REGISTRY)/$*:amd64-$(IMAGE_TAG) $*
+	docker build $(DOCKER_BUILD_FLAGS) -t $(DOCKER_REGISTRY)/$*:arm64v8-$(IMAGE_TAG) -f $*/Dockerfile.arm64v8 $*
 
 # You must be logged into DOCKER_REGISTRY before you can push.
 .PHONY: docker-push
 docker-push: $(addsuffix -push,$(IMAGES))
 
 %-push:
-	docker push $(DOCKER_REGISTRY)/$*:$(IMAGE_TAG)
+	docker push $(DOCKER_REGISTRY)/$*:amd64-$(IMAGE_TAG)
+	docker push $(DOCKER_REGISTRY)/$*:arm64v8-$(IMAGE_TAG)
+
+	# Manifest creation
+	@if [ -d $(HOME)/.docker/manifests/docker.io_$(DOCKER_REGISTRY)_$*-$(IMAGE_VERSION) ]; then \
+		rm -rf $(HOME)/.docker/manifests/docker.io_$(DOCKER_REGISTRY)_$*-$(IMAGE_VERSION); \
+	fi
+
+	docker manifest create $(DOCKER_REGISTRY)/$*:$(IMAGE_TAG) $(DOCKER_REGISTRY)/$*:amd64-$(IMAGE_TAG) $(DOCKER_REGISTRY)/$*:arm64v8-$(IMAGE_TAG)
```

Comment:
This manifest feature is really cool -- thanks for sparking my learning on it.

Although the manifest creation/annotation/push works for my Dockerhub and ACR registries, I see the following when attempting to push to my `quay.io` registry:
```
$ make docker-push
...
# Manifest creation
docker manifest create quay.io/vdice/brigade-api:v0.20.0-15-gc1af200 quay.io/vdice/brigade-api:amd64-v0.20.0-15-gc1af200 quay.io/vdice/brigade-api:arm64v8-v0.20.0-15-gc1af200
unsupported manifest media type and no default available: application/vnd.docker.distribution.manifest.v1+prettyjws
make: *** [brigade-api-push] Error 1
```

Wondering if there might be other registries where manifest is not yet supported?  (I'll dig into Quay.io's status here.)

If so, I'm wondering if we can come up with an approach where `docker-push` will still work for users pushing to registries without manifest support?

Reply:
> To invoke `docker manifest` commands, I had to add `"experimental": "enabled"` to my cli config at `~/.docker/config.json` -- we might want to add a comment/note here or somehow inform the user if this target errors out due to the lack of this flag in the user's cli config.

This is fairly easily detected and automatically ajusted. It would even be possible to set experimental: true, run the manifest commands, then set it back to what it used to be using `jq` for instance. I build some other multi-arch images on travis/azure pipelines using this technique for instance.

> Although the manifest creation/annotation/push works for my Dockerhub and ACR registries, I see the following when attempting to push to my `quay.io` registry:

I thought brigade was just using docker hub, so I didn't think the manifest usage would be a problem. However, achieving support for quay etc. should be pretty straight forward, just use a toggle `USE_MANIFEST` or whatever, and modify what push does. I can definitely add this to the makefile. Wrt quay specifically, they are apparently working on schema v2 support: https://github.com/moby/buildkit/issues/409#issuecomment-394757219, but no updates (at least in that thread) for months.

For the simplest solution, simply not creating/pushing the manifest lists would enable quay support. It would however force people to modify helm chart values before installing, but I guess that would be the case regardless, given that they are using their own registry.

Reply:
> For the simplest solution, simply not creating/pushing the manifest lists would enable quay support. It would however force people to modify helm chart values before installing, but I guess that would be the case regardless, given that they are using their own registry.

Good point -- perhaps, if the user needs to update tag values anyways, supporting quay should be as straightforward as configuring the manifest commands to run optionally.  (Thanks for that related issue link.)

I'd be fine with either the `USE_MANIFEST` approach for conditionalizing the push command or with `%-docker-push` only issuing `docker push` commands and maybe a separate `%-docker-manifest` command to wrap the manifest commands.  Then, CI (or anyone) can run `make docker-push docker-manifest` to push and create/update manifests for all images.  Will leave it up to you.

Reply:
> This is fairly easily detected and automatically ajusted. It would even be possible to set experimental: true, run the manifest commands, then set it back to what it used to be using `jq` for instance. I build some other multi-arch images on travis/azure pipelines using this technique for instance.

I'd be down to add this logic as a special target, for use in CI.  Not sure I'd want it to run automagically by default on user's machines, though.